RevenueCatUI/UIKit/PaywallViewController.swift

```diff
@@ -82,12 +86,24 @@ public class PaywallViewController: UIViewController {
         self.addChild(self.hostingController)
         self.view.addSubview(self.hostingController.view)
         self.hostingController.didMove(toParent: self)
-    }
-
-    public override func viewWillLayoutSubviews() {
-        super.viewWillLayoutSubviews()
+        hostingController.view.translatesAutoresizingMaskIntoConstraints = false
+
+        NSLayoutConstraint.activate([
+            hostingController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            hostingController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            hostingController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ])
+
+        // for footers we only want to tie to the sides and the bottom.
+        if self.mode != .footer && self.mode != .condensedFooter {
```

Comment:
Would it be easier to check if it's full screen instead?

Reply:
That was my original code, and yeah, it looks simpler. 

I modified it to this because I felt like it was more "correct" to say that footers don't need to be constrained to the top, rather than full screen being the special case. Like, if we had yet another type (i.e.: banner, badge, etc), then we'd have to also special case those. 

But that's super easy to refactor again in the future, and we don't know as of right now what types we'll have, so I'll go with checking only full screen for now. Updating